docs/SUBMISSION.md

```diff
@@ -39,16 +51,27 @@ You will find the implementation file at [`/services/submission/src/main/java/ap
 
-Constraints maintained as enviroment variables which are present as secrets in the Vault /cwa-server/submission
+Some constraints are maintained as enviroment variables and kept in Vault under the key path: /cwa-server/submission
 
 The constraints put on submitted TEK's are as follows:
 
-* Each TEK contains a `StartIntervalNumber` (a date e.g. 2nd July 2020)
-* The period covered by the data file must not exceed the configured maximum number of days, represented by the `MAX_NUMBER_OF_KEYS` property which is in the vault.
-* The total combined rolling period for a single TEK cannot exceed maximum rolling period, represented by the `MAX_ROLLING_PERIOD` property which is in the vault.
-* More than one TEK with the same `StartIntervalNumber` may be submitted, these will have their rolling period's combined.
+* Each TEK contains a `StartIntervalNumber` with the value set at midnight (a datetime e.g. 2nd July 2020 00:00)
+* The number of TEK's must not exceed the configured maximum number of keys, represented by the `MAX_NUMBER_OF_KEYS` property which is in the vault
+* More than one TEK with the same `StartIntervalNumber` may be submitted, these will have their rolling period's combined
+* Each TEK must carry either the information of transmission risk level or days since onset of symptoms. If one is missing, the other can be derived. If a TEK is missing both values or contains values which are not in the allowed ranges for each field the payload shall be rejected
+* The origin country from the submission payload must be a valid ISO country code and must either be one of the supported countries (maintained in Vault) or it must be empty, in which case it will be defaulted to DE
+* The visited countries list from the submission payload must either contain ISO country codes which are part of the supported countries or it must be an empty list, in which case it will be prefilled with the default origin country (e.g. DE)
+
+There are other validations performed prior to persisting keys, which check whether specific fields are in acceptable ranges as defined by the 
+system as well as GAEN specification. For this purpose the Java Bean Validation 2.0 framework is used at the entity level. Please see 
+[DiagnosisKey](/common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKey.java) for the complete list.
+
+## Field Derivations
+
+Starting with version 1.5 the mobile client stops sending transmission risk level (TRL) values but includes a new field that is used to describe the infectiousness of a person called 'days since onset of symptoms' (DSOS). To ensure backward compatibility with older clients when distributing keys submitted by newer clients, the server has to derive the missing TRL field from the DSOS value using a mapping maintained in the application configuration. Similarly, to ensure forward compatibility when distributing keys submitted by older clients ( version 1.4 or less), the server
```

Comment:
```suggestion
Starting with version 1.5 the mobile client stops sending transmission risk level (TRL) values but includes a new field that is used to describe the infectiousness of a person called 'days since onset of symptoms' (DSOS). To ensure backward compatibility with older clients when distributing keys submitted by newer clients, the server has to derive the missing TRL field from the DSOS value using a mapping maintained in the application configuration. Similarly, to ensure forward compatibility when distributing keys submitted by older clients (version 1.4 or less), the server
```